tmail-backend/mailbox/encrypted/memory/src/main/java/com/linagora/tmail/encrypted/InMemoryEncryptedEmailContentStore.scala

```diff
@@ -0,0 +1,78 @@
+package com.linagora.tmail.encrypted
+
+import com.google.common.base.Preconditions
+import com.google.inject.{AbstractModule, Scopes}
+import com.linagora.tmail.encrypted.EncryptedEmailContentStore.POSITION_NUMBER_START_AT
+import com.linagora.tmail.encrypted.InMemoryEncryptedEmailContentStore.{emailContentStore, messageIdBlobIdStore}
+import org.apache.james.blob.api.BlobStore.StoragePolicy
+import org.apache.james.blob.api.{BlobId, BlobStore, BucketName}
+import org.apache.james.mailbox.model.MessageId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.lang
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicInteger
+import javax.inject.Inject
+
+case class EncryptedEmailContentStoreMemoryModule() extends AbstractModule {
+  override def configure(): Unit = {
+    bind(classOf[InMemoryEncryptedEmailContentStore]).in(Scopes.SINGLETON)
+
+    bind(classOf[EncryptedEmailContentStore]).to(classOf[InMemoryEncryptedEmailContentStore])
+  }
+}
+
+object InMemoryEncryptedEmailContentStore {
+  val emailContentStore: scala.collection.concurrent.Map[MessageId, EncryptedEmailContent] = scala.collection.concurrent.TrieMap()
+  val messageIdBlobIdStore: scala.collection.concurrent.Map[MessageId, Map[Int, BlobId]] = scala.collection.concurrent.TrieMap()
+}
+
+class InMemoryEncryptedEmailContentStore @Inject()(blobStore: BlobStore,
+                                                   bucketName: BucketName,
+                                                   storagePolicy: StoragePolicy) extends EncryptedEmailContentStore {
+
+  override def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Publisher[Unit] =
+    storeAttachment(messageId, encryptedEmailContent.encryptedAttachmentContents)
+      .`then`(SMono.just(emailContentStore.put(messageId, encryptedEmailContent)))
+
+  override def delete(messageId: MessageId): Publisher[Unit] =
+    removeAttachment(messageId)
+      .`then`(SMono.just(emailContentStore.remove(messageId)))
+
+  override def retrieveFastView(messageId: MessageId): Publisher[EncryptedEmailFastView] =
+    SMono.just(emailContentStore.get(messageId)
+      .map(encryptedEmailContent => EncryptedEmailFastView.from(encryptedEmailContent))
+      .getOrElse(throw MessageNotFoundException(messageId)))
```

Comment:
Can we avoid this throw?

Reply:
We can use `Either` or `Try`, but I don't want to use both in interface class (trait), maybe in the future, we can have one java implement it. 
So return `null` is acceptable?

Reply:
or `SMono.error`?